src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -282,6 +304,10 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
     );
   }
 
+  function shouldRenderQueryEditorExtensions(): boolean {
+    return Boolean(queryEditorExtensionMap && Object.keys(queryEditorExtensionMap).length);
```

Comment:
Do we trust that `queryEditorExtensionMap` is going to be a plain object? I am not sure where it is coming from and hence asking: could it possibly be manipulated in the metadata store?

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/7f0e39eb9809c95b98069cc971611edc2cbbc62b/src/plugins/data/public/ui/ui_service.ts#L31

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/7f0e39eb9809c95b98069cc971611edc2cbbc62b/src/plugins/data/public/ui/ui_service.ts#L48-L51

what is the metadata store?